fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1108,15 +1105,25 @@ struct TransactionRateInfo {
 	TransactionRateInfo(double rate) : rate(rate), limit(0) {}
 
 	void reset(double elapsed) {
-		limit = std::min(0.0,limit) + std::min(rate * elapsed, SERVER_KNOBS->START_TRANSACTION_MAX_TRANSACTIONS_TO_START);
+		limit = std::min(0.0, limit) + rate * elapsed; // Adjust the limit based on the full elapsed interval in order to properly erase a deficit
+		limit = std::min(limit, rate * SERVER_KNOBS->START_TRANSACTION_BATCH_INTERVAL_MAX); // Don't allow the rate to exceed what would be allowed in the maximum batch interval
```

Comment:
Is this necessary? It seems like `elapsed` will not be larger than `SERVER_KNOBS->START_TRANSACTION_BATCH_INTERVAL_MAX` and thus `std::min(0.0, limit) + rate * elapsed` will always be a value smaller or equal to `rate * SERVER_KNOBS->START_TRANSACTION_BATCH_INTERVAL_MAX`.

Reply:
Oh never mind, I didn't see the change below.